### PR TITLE
Create an `alpha` version of `azure.yaml` schema

### LINKED ISSUE #1925: [Issue] Azure.yaml - terraform is not a valid value
- [X] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**

azd version 0.8.0-beta.1 (commit 08209da7c22d28c6a123ef4e88d433132066b800)

**Describe the bug**

The JSON schema backing the `azure.yaml` (<https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/v1.0/azure.yaml.json>)was changed with version 0.8.0 of `azd` restricting the allowed values for the `infra.provider` to `bicep`.

Although the provisoning via Terraform became a alpha feature that needs to be explicitly enabled, the JSON schema should either allow this value or a new version of the JSON schema should be created that is shipped with the templates. 

**To Reproduce**

- Execute `azd init -t Azure-Samples/todo-nodejs-mongo-terraform`
- Open the cloned project in VSCode
- Open the `azure.ymal` file

=> you will find a red curly line underlining the provider value `terraform` 

**Expected behavior**

As mentioned above:

- Either the Terraform provider is a valid value again for the enum in the infra section of the `azure.yaml`
- A new version of the JSON schema is released that is not containing Terraform. However the original version should still contain the Terraform as a valid value and should then be referenced by the "old" Terraform templates

**Environment**

n/a

**Additional context**

n/a

### PR DESCRIPTION
**Pull request by ellismg**
We don't want to include any values in the schema that are tied to alpha features in the `v1.0` schema, since we may remove or change these values during development.

This change introduces a new copy of this schema and adds "terraform" as a supported member of an enum (this is a new value that is supported behind an alpha feature)

Fixes #1925

### CODE REVIEW

File: schemas/alpha/azure.yaml.json [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+                        "type": "string",
+                        "title": "Type of Azure resource used for service implementation",
+                        "description": "If omitted, App Service will be assumed.",
+                        "enum": [
+                            "",
+                            "appservice",
+                            "containerapp",
+                            "function",
+                            "staticwebapp",
+                            "aks"
```
Comment (rajeshkamal5050) [inline schemas/alpha/azure.yaml.json:92]: Can you also add springapp?

**Review by rajeshkamal5050 — CHANGES_REQUESTED** [review]

**Comment by savannahostrowski** [comment]
Should we also update the TF todo templates with the alpha schema as part of this PR?

**Comment by ellismg** [comment]
Pushed [3aaaa3c](https://github.com/Azure/azure-dev/pull/2251/commits/3aaaa3c8c4c0e1e47fcbfe4c6e5e797269332306) to reference these in the bicep templates, @savannahostrowski.

**Review by savannahostrowski — APPROVED** [review]
LGTM! We should also probably let the ASA team know to update their template as well!

cc:@rajeshkamal5050

**Review by rajeshkamal5050 — APPROVED** [review]

**Pull request merged**